test/elixir_analyzer/exercise_test/common_checks/private_helper_functions_test.exs

```diff
@@ -0,0 +1,452 @@
+                else
+                  unquote(expr)
+                end
+              end
+            end
+
+            def calculate(1), do: 1
+
+            def calculate(radicand) do
+              guess = div(radicand, 2)
+              do_calculate(radicand, guess)
+            end
+
+            defp do_calculate(radicand, guess) when is_positive(guess) do
+              new_guess = div(guess + div(radicand, guess), 2)
+
+              unless_equal_to new_guess, guess do
+                do_calculate(radicand, new_guess)
+              end
+            end
+          end
+        end
+
+      assert PrivateHelperFunctions.run(code, @sqrt_stub) == []
+    end
+
+    test "solution with public guard" do
+      code =
+        quote do
+          defmodule SquareRoot do
+            defguard is_positive(n) when n >= 0
+
+            defmacrop unless_equal_to(guess, goal, do: expr) do
+              quote do
+                if unquote(guess) == unquote(goal) do
+                  unquote(goal)
+                else
+                  unquote(expr)
+                end
+              end
+            end
+
+            def calculate(1), do: 1
+
+            def calculate(radicand) do
+              guess = div(radicand, 2)
+              do_calculate(radicand, guess)
+            end
+
+            defp do_calculate(radicand, guess) when is_positive(guess) do
+              new_guess = div(guess + div(radicand, guess), 2)
+
+              unless_equal_to new_guess, guess do
+                do_calculate(radicand, new_guess)
+              end
+            end
+          end
+        end
+
+      comment = %{
+        @comment
+        | params: %{actual: "defguard is_positive(a)", expected: "defguardp is_positive(a)"}
+      }
+
+      assert PrivateHelperFunctions.run(code, @sqrt_stub) == [{:fail, comment}]
+    end
+
+    test "solution with public macro" do
+      code =
+        quote do
+          defmodule SquareRoot do
+            defguardp is_positive(n) when n >= 0
+
+            defmacro unless_equal_to(guess, goal, do: expr) do
+              quote do
+                if unquote(guess) == unquote(goal) do
+                  unquote(goal)
+                else
+                  unquote(expr)
+                end
+              end
+            end
+
+            def calculate(1), do: 1
+
+            def calculate(radicand) do
+              guess = div(radicand, 2)
+              do_calculate(radicand, guess)
+            end
+
+            defp do_calculate(radicand, guess) when is_positive(guess) do
+              new_guess = div(guess + div(radicand, guess), 2)
+
+              unless_equal_to new_guess, guess do
+                do_calculate(radicand, new_guess)
+              end
+            end
+          end
+        end
+
+      comment = %{
+        @comment
+        | params: %{
+            actual: "defmacro unless_equal_to(a, b, c)",
+            expected: "defmacrop unless_equal_to(a, b, c)"
```

Comment:
I understand why we don't want to recreate the actual arguments that were given (might be a complex pattern matching expression, might get misformatted, it will distract from the point of the check). I also understand that showing the argument list at all is necessary to reveal information about the arity. But what would you think about skipping names altogether?

```suggestion
            actual: "defmacro unless_equal_to(_, _, _)",
            expected: "defmacrop unless_equal_to(_, _, _)"
```

Reply:
Ah that's a good idea, I didn't think about that!
I went back and forth between using `a, b, c` or the real arguments, but this is better.